### PR TITLE
python3Packages.pycec: init at 0.5.1

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Python modules to access HDMI CEC devices

https://github.com/konikvranik/pycec/

`pyCEC` is a Home Assistant dependency.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libcec/default.nix [hunk 1]
```diff
@@ -1,33 +1,64 @@
...
+    };
+  };
 
-  # Fix dlopen path
-  patchPhase = ''
-    substituteInPlace include/cecloader.h --replace "libcec.so" "$out/lib/libcec.so"
-  '';
+in
+rec {
+  libcec_4 = generic {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libcec/default.nix:55]: Generics are usually split of into another file.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113842` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
21 packages built:
</summary>
<ul>
<li>kodi</li>
<li>kodi-gbm</li>
<li>kodi-wayland</li>
<li>kodiGBM</li>
<li>kodiPlain (kodiPlugins.kodi)</li>
<li>kodiPlainWayland</li>
<li>kodiPlugins.inputstream-adaptive</li>
<li>kodiPlugins.joystick</li>
<li>kodiPlugins.kodi-platform</li>
<li>kodiPlugins.pvr-hdhomerun</li>
<li>kodiPlugins.pvr-hts</li>
<li>kodiPlugins.pvr-iptvsimple</li>
<li>kodiPlugins.steam-controller</li>
<li>kodiPlugins.vfs-libarchive</li>
<li>kodiPlugins.vfs-sftp</li>
<li>libcec (libcec_4)</li>
<li>libcec_6</li>
<li>moonlight-embedded</li>
<li>python37Packages.pycec</li>
<li>python38Packages.pycec</li>
<li>python39Packages.pycec</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
kodiPlain:
<details>


`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/video/kodi/default.nix:288:7:

```
    |
288 |       license     = licenses.gpl2;
    |       ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
unzip is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/video/kodi/default.nix:159:5:

```
    |
159 |     buildInputs = [
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
zip is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/video/kodi/default.nix:159:5:

```
    |
159 |     buildInputs = [
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
Unused argument: `fetchpatch`.
Near pkgs/applications/video/kodi/default.nix:1:16:

```
  |
1 | { stdenv, lib, fetchpatch, fetchurl, fetchFromGitHub, autoconf, automake, libtool, makeWrapper, linuxHeaders
  |                ^
```
Unused argument: `autoreconfHook`.
Near pkgs/applications/video/kodi/default.nix:4:23:

```
  |
4 | , boost, avahi, lame, autoreconfHook
  |                       ^
```
Unused argument: `glew`.
Near pkgs/applications/video/kodi/default.nix:8:27:

```
  |
8 | , alsaLib, libGLU, libGL, glew, fontconfig, freetype, ftgl
  |                           ^
```
</details>
kodiPlugins.inputstream-adaptive:
<details>


Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/video/kodi/plugins.nix:466:17:

```
    |
466 |     patches = [ ./yatp/dont-monkey.patch ];
    |                 ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
kodiPlugins.joystick:
<details>


Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/video/kodi/plugins.nix:466:17:

```
    |
466 |     patches = [ ./yatp/dont-monkey.patch ];
    |                 ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
kodiPlugins.kodi-platform:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/video/kodi/plugins.nix:51:5:

```
   |
51 |     buildInputs = [ kodiPlain libcec_platform tinyxml ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Package is missing a license

Near pkgs/applications/video/kodi/plugins.nix:51:5:

```
   |
51 |     buildInputs = [ kodiPlain libcec_platform tinyxml ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/video/kodi/plugins.nix:466:17:

```
    |
466 |     patches = [ ./yatp/dont-monkey.patch ];
    |                 ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
kodiPlugins.pvr-hdhomerun:
<details>


Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/video/kodi/plugins.nix:466:17:

```
    |
466 |     patches = [ ./yatp/dont-monkey.patch ];
    |                 ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
kodiPlugins.pvr-hts:
<details>


Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/video/kodi/plugins.nix:466:17:

```
    |
466 |     patches = [ ./yatp/dont-monkey.patch ];
    |                 ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
kodiPlugins.pvr-iptvsimple:
<details>


Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/video/kodi/plugins.nix:466:17:

```
    |
466 |     patches = [ ./yatp/dont-monkey.patch ];
    |                 ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
kodiPlugins.steam-controller:
<details>


Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/video/kodi/plugins.nix:466:17:

```
    |
466 |     patches = [ ./yatp/dont-monkey.patch ];
    |                 ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
kodiPlugins.vfs-libarchive:
<details>


Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/video/kodi/plugins.nix:466:17:

```
    |
466 |     patches = [ ./yatp/dont-monkey.patch ];
    |                 ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
kodiPlugins.vfs-sftp:
<details>


Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/video/kodi/plugins.nix:466:17:

```
    |
466 |     patches = [ ./yatp/dont-monkey.patch ];
    |                 ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
moonlight-embedded:
<details>


`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/misc/moonlight-embedded/default.nix:31:5:

```
   |
31 |     license = licenses.gpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
pkg-config is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/misc/moonlight-embedded/default.nix:22:3:

```
   |
22 |   buildInputs = [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
</details>
</details>

File: pkgs/development/libraries/libcec/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/libraries/libcec/default.nix:55]: I guess then that `libwebsockets` is out-dated.

**Comment by aanderse** [comment]
In a day or two we won't need this anymore. `kodi` `19.x` has been released and [updated in our repos](https://github.com/NixOS/nixpkgs/pull/105417), merge pending.

Can we please bump [this PR](https://github.com/NixOS/nixpkgs/pull/105003) instead?

Am I missing anything, or does this sound like a good plan?

**Comment by fabaff** [comment]
> Can we please bump this PR instead?

Yes, we should in that case.

> Am I missing anything, or does this sound like a good plan?

I don't think so. It doesn't make sense to keep 4.0.7 if it's not used any longer and Kodi 18.x seems to be the only package that would it require.

**Comment by aanderse** [comment]
Awesome. Thanks.

I'm write to some release notes for `kodi` today and I think we are good to merge.

**Review by mweinelt — APPROVED** [review]

**Pull request merged**